File: csharp/004/004.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Euler {
    static public void Main() {
        List<ulong> palindromes = new List<ulong>();

        for (ulong x = 100; x <= 999; x++) {
            for (ulong y = 100; y <= 999; y++) {
                ulong product = x * y;

                String s = product.ToString();

                if (s == String.Concat(Enumerable.Reverse(s))) {
                    palindromes.Add(product);
                }
            }
        }

        Console.WriteLine("{0}", palindromes.Max(x => x));
    }
}

File: lua/004/004.lua
table.max = function(t)
  if #t == 0 then
    return nil
  end

  local m = t[1]

  for i = 2, #t do
    if t[i] > m then
      m = t[i]
    end
  end

  return m
end

palindromes = {}

for x = 100, 999, 1 do
  for y = 100, 999, 1 do
    product = x * y

    s = tostring(product)

    if s == string.reverse(s) then
      table.insert(palindromes, product)
    end
  end
end

print(table.max(palindromes))
